File: typescript/src/ancilis/aksi/identifiers.ts
/** AKSI control identifier boundaries. */

export const AKSI_PREFIX = "AKSI-";
const AKSI_LEGACY_PREFIX = "AKSI_";

export function isPrefixed(controlId: string): boolean {
  return controlId.startsWith(AKSI_PREFIX) || controlId.startsWith(AKSI_LEGACY_PREFIX);
}

export const is_prefixed = isPrefixed;

export function unprefix(controlId: string): string {
  const normalized = controlId.startsWith(AKSI_LEGACY_PREFIX)
    ? `${AKSI_PREFIX}${controlId.slice(AKSI_LEGACY_PREFIX.length)}`
    : controlId;
  return normalized.startsWith(AKSI_PREFIX)
    ? normalized.slice(AKSI_PREFIX.length)
    : normalized;
}

export function prefix(controlId: string): string {
  return `${AKSI_PREFIX}${unprefix(controlId)}`;
}

File: python/src/ancilis/aksi/identifiers.py
"""AKSI control identifier boundaries."""

AKSI_PREFIX = "AKSI-"
AKSI_LEGACY_PREFIX = "AKSI_"


def is_prefixed(control_id: str) -> bool:
    return control_id.startswith(AKSI_PREFIX) or control_id.startswith(AKSI_LEGACY_PREFIX)


def unprefix(control_id: str) -> str:
    if control_id.startswith(AKSI_LEGACY_PREFIX):
        return control_id[len(AKSI_LEGACY_PREFIX) :]
    return control_id.removeprefix(AKSI_PREFIX)


def prefix(control_id: str) -> str:
    return f"{AKSI_PREFIX}{unprefix(control_id)}"
